gpt4all-chat/chat.cpp

```diff
@@ -424,13 +424,6 @@ bool Chat::deserialize(QDataStream &stream, int version)
     if (!m_modelInfo.id().isEmpty())
         emit modelInfoChanged();
 
-    bool discardKV = m_modelInfo.id().isEmpty();
```

Comment:
Why do we no longer need to discard the KV cache when the id is empty? There's no comment here so I'm not entirely sure what that means anyway. It was added in f0742c22f4348e675ae86767f426a20867d7ee93.

Reply:
Ah, maybe I was too hasty. Will go and look.